subsys/bluetooth/host/smp.c

```diff
@@ -4048,16 +4048,24 @@ static uint8_t smp_id_add_replace(struct bt_smp *smp, struct bt_keys *new_bond)
 	 */
 	__ASSERT_NO_MSG(!(smp->remote_dist & BT_SMP_DIST_ID_KEY));
 
-	conflict = bt_id_find_conflict(new_bond);
+	conflict = bt_id_find_conflict(new_bond, IS_ENABLED(CONFIG_BT_ID_AUTO_SWAP_MATCHING_BONDS));
 	if (conflict) {
 		LOG_DBG("New bond conflicts with a bond on id %d.", conflict->id);
 	}
 
-	if (conflict && !IS_ENABLED(CONFIG_BT_ID_UNPAIR_MATCHING_BONDS)) {
+	if (conflict && !IS_ENABLED(CONFIG_BT_ID_AUTO_SWAP_MATCHING_BONDS) &&
+	    !IS_ENABLED(CONFIG_BT_ID_UNPAIR_MATCHING_BONDS)) {
 		LOG_WRN("Refusing new pairing. The old bond must be unpaired first.");
```

Comment:
LOG_ERR?

Reply:
This is upstream warning. Maybe let it stay as is?